packages/psss/src/views/Tabs/ProfileTabView.js

```diff
@@ -34,38 +36,82 @@ const SuccessMessage = styled(SmallAlert)`
   margin-bottom: 1.5rem;
 `;
 
+const Divider = styled(MuiDivider)`
+  margin: 1.8rem 0;
+`;
+
+const STATUS = {
+  IDLE: 'idle',
+  LOADING: 'loading',
+  SUCCESS: 'success',
+  ERROR: 'error',
+  DISABLED: 'disabled',
+};
+
 const ProfileTabViewComponent = React.memo(({ user, onUpdateProfile }) => {
+  const [status, setStatus] = useState(STATUS.IDLE);
   const [errorMessage, setErrorMessage] = useState(null);
   const [successMessage, setSuccessMessage] = useState(null);
-  const [isLoading, setIsLoading] = useState(false);
+  const [profileImage, setProfileImage] = useState({
+    fileId: null,
+    data: user.profileImage,
+  });
   const { handleSubmit, register, errors } = useForm();
 
   const onSubmit = handleSubmit(async ({ firstName, lastName, role, employer }) => {
-    setIsLoading(true);
+    setStatus(STATUS.LOADING);
     setErrorMessage(null);
     try {
       await onUpdateProfile({
         first_name: firstName,
         last_name: lastName,
         position: role,
         employer,
+        profileImage: profileImage.fileId && profileImage,
       });
-      setIsLoading(false);
+      setStatus(STATUS.SUCCESS);
       setSuccessMessage('Profile successfully updated.');
     } catch (error) {
-      setIsLoading(false);
+      setStatus(STATUS.ERROR);
       setErrorMessage(error.message);
     }
   });
 
+  const handleFileChange = async event => {
+    setStatus(STATUS.DISABLED);
+    const fileObject = event.target.files[0];
+    const base64 = await createBase64Image(fileObject);
+    const fileName = fileObject.name.replace(/\.[^/.]+$/, '');
+    setProfileImage({
+      fileId: `${user.id}-${fileName}`,
+      data: base64,
+    });
+    setStatus(STATUS.IDLE);
+  };
+
+  const handleFileDelete = () => {
+    // We need to set a file id to something so that the backend will pick up the profileImage field
+    // as an updatedField and save the change.
+    setProfileImage({ fileId: '-', data: null });
```

Comment:
```suggestion
    setProfileImage({ fileId: '', data: null });
```
Does that work too?

Reply:
no unfortunately not. In the submit handler, I use the fileId to determine whether or not the profileImage is updated. See line 70. I could tweak that if you think it is confusing.

Reply:
Yeah I do find it a little confusing - could we check if `profileImage.data !== user.profileImage`?